src/edges.rs

```diff
@@ -0,0 +1,131 @@
+use image::{Pixel, GenericImage, ImageBuffer, Luma};
+use image::imageops::blur;
+use gradients::{vertical_sobel, horizontal_sobel};
+
+/// Runs the canny edge detection algorithm on the provided `ImageBuffer`.
+///
+/// Returns a binary image, where edge pixels have a value of 255 and non-edge pixels a value of 0.
+pub fn canny<I>(image: &I,
+                low_threshold: u8,
+                high_threshold: u8)
+                -> ImageBuffer<Luma<u8>, Vec<<Luma<u8> as Pixel>::Subpixel>>
+    where I: GenericImage<Pixel = Luma<u8>> + 'static
```

Comment:
The return type can be written more concisely as ImageBuffer<Luma<u8>, Vec<u8>>.